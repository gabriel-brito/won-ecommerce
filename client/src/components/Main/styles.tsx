import styled from 'styled-components'

export const Wrapper = styled.main`
  align-items: center;
  background-color: #06092b;
  color: #fff;
  display: flex;
  flex-direction: column;
  height: 100%;
  justify-content: center;
  padding: 3rem;
  text-align: center;
  width: 100%;
`

export const Logo = styled.img`
  margin-bottom: 2rem;
  width: 25rem;
`

export const Title = styled.h1`
  font-size: 2.5rem;
`

export const Description = styled.h2`
  font-size: 2rem;
  font-weight: 400;
`

export const Illustration = styled.img`
  margin-top: 3rem;
  width: min(30rem, 100%);
`
