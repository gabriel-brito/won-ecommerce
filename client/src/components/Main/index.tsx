import * as S from './styles'

const Main = () => (
  <S.Wrapper>
    <S.Logo
      src="/img/logo.svg"
      alt="Image of an atom and React Avançado written on the side."
    />
    <S.Title>React Avançado</S.Title>
    <S.Description>
      TypeScript, ReactJS, NextJS and Styled Components
    </S.Description>
    <S.Illustration
      src="/img/hero-illustration.svg"
      alt="A developer looking to a screen with code."
    />
  </S.Wrapper>
)


export default Main
