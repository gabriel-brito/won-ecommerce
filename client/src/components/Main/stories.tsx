import { withKnobs, text } from '@storybook/addon-knobs'
import { Story, Meta } from '@storybook/react/types-6-0'
import Main from '.'

export default {
  title: 'Main',
  component: Main,
  decorators: [withKnobs]
} as Meta

export const Basic: Story = () => (
  <Main
    title={text('Text', 'React AvanÃ§ado')}
    description={text(
      'decription',
      'TypeScript, ReactJS, NextJS and Styled Components'
    )}
  />
)
