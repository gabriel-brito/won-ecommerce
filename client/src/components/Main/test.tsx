import { screen, render } from '@testing-library/react'
import Main from './index'

describe('<Main />', () => {
  it('Should Main File render', () => {
    const { container } = render(<Main />)

    expect(
      screen.getByRole('heading', { name: /React avanÃ§ado/i })
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('Should have background-color in it', () => {
    const { container } = render(<Main />);

    expect(container.firstChild).toHaveStyle({ 'background-color': '#06092b' });
  });
})
