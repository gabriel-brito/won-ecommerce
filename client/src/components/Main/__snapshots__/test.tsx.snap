// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> Should Main File render 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #06092b;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 3rem;
  text-align: center;
  width: 100%;
}

.c1 {
  margin-bottom: 2rem;
  width: 25rem;
}

.c2 {
  font-size: 2.5rem;
}

.c3 {
  font-size: 2rem;
  font-weight: 400;
}

.c4 {
  margin-top: 3rem;
  width: min(30rem,100%);
}

<main
  class="c0"
>
  <img
    alt="Image of an atom and React Avançado written on the side."
    class="c1"
    src="/img/logo.svg"
  />
  <h1
    class="c2"
  >
    React Avançado
  </h1>
  <h2
    class="c3"
  >
    TypeScript, ReactJS, NextJS and Styled Components
  </h2>
  <img
    alt="A developer looking to a screen with code."
    class="c4"
    src="/img/hero-illustration.svg"
  />
</main>
`;
